doc/man3/BIO_s_mem.pod

```diff
@@ -104,6 +105,15 @@ BIO is set to BIO_NOCLOSE, before freeing the BUF_MEM the data pointer
 in it must be set to NULL as the data pointer does not point to an
 allocated memory.
 
+Calling BIO_reset() on a read write memory BIO with BIO_FLAGS_NONCLEAR_RST
+flag set can have unexpected outcome when the reads and writes to the
+BIO are intertwined. As documented above the BIO will be reset to the
+state after the last completed write operation. The effects of reads
+preceeding that write operation cannot be undone.
+
+Calling BIO_get_mem_ptr() prior to a BIO_reset() call with
+BIO_FLAGS_NONCLEAR_RST set has the same effect as a write operation.
```

Comment:
should we note that BIO_get_mem_data does not have the same effect?

Reply:
I do not see it as necessary. It is implied by not mentioning it here. But if you think it is necessary to mention it explicitly please tell me and I can add it.

Reply:
ok, fine.